locked-nft/contracts/NFTLocker.cdc

```diff
@@ -123,52 +277,93 @@ access(all) contract NFTLocker {
         access(all) view fun getIDs(nftType: Type): [UInt64]?
         access(Operate) fun lock(token: @{NonFungibleToken.NFT}, duration: UInt64)
         access(Operate) fun unlock(id: UInt64, nftType: Type): @{NonFungibleToken.NFT}
+        access(Operate) fun unlockWithAuthorizedDeposit(
+            id: UInt64,
+            nftType: Type,
+            receiverName: String,
+            passThruParams: {String: AnyStruct}
+        )
     }
 
     /// Deprecated in favor of Operate entitlement
     ///
     access(all) resource interface LockProvider: LockedCollection {}
 
-    /// Entitlement that grants the ability to operate the NFTLocker Collection
-    access(all) entitlement Operate
-
     /// An NFT Collection
     ///
     access(all) resource Collection: LockedCollection, LockProvider {
+        /// Locked NFTs
+        ///
         access(all) var lockedNFTs: @{Type: {UInt64: {NonFungibleToken.NFT}}}
 
         /// Unlock an NFT of a given type
         ///
         access(Operate) fun unlock(id: UInt64, nftType: Type): @{NonFungibleToken.NFT} {
             pre {
-                NFTLocker.canUnlockToken(
-                    id: id,
-                    nftType: nftType
-                ) == true : "locked duration has not been met"
+                NFTLocker.canUnlockToken(id: id, nftType: nftType): "locked duration has not been met"
             }
 
+            // Get the locked token
             let token <- self.lockedNFTs[nftType]?.remove(key: id)!!
 
+            // Remove the locked data
             if let lockedTokens = &NFTLocker.lockedTokens[nftType] as auth(Remove) &{UInt64: NFTLocker.LockedData}? {
                 lockedTokens.remove(key: id)
             }
-            NFTLocker.totalLockedTokens = NFTLocker.totalLockedTokens - 1
 
-            emit NFTUnlocked(
-                id: token.id,
-                from: self.owner?.address,
-                nftType: nftType
-            )
+            // Decrement the total locked tokens
+            NFTLocker.totalLockedTokens = NFTLocker.totalLockedTokens - 1
 
+            // Emit events
+            emit NFTUnlocked(id: token.id, from: self.owner?.address, nftType: nftType)
             emit Withdraw(id: token.id, from: self.owner?.address)
 
             return <-token
         }
 
+        /// Force unlock the NFT with the given id and type, and deposit it using the receiver's deposit method;
+        /// additional function parameters may be required by the receiver's deposit method and are passed in the
+        /// passThruParams map.
+        ///
+        access(Operate) fun unlockWithAuthorizedDeposit(
```

Comment:
unlock a locked NFT if an authorized deposit method has been defined (e.g., deposit into escrow leaderboard) - no need for admin intervention